lisa/microsoft/testsuites/power/common.py

```diff
@@ -49,6 +50,59 @@ def is_distro_supported(node: Node) -> None:
         )
 
 
+def check_hibernation_disk_requirements(node: Node) -> None:
+    """
+    Check if the VM has sufficient disk space for hibernation.
+    Hibernation requires disk space based on RAM size using the formula:
+    - 2 × RAM if RAM ≤ 8 GB
+    - 1.5 × RAM if 8 < RAM ≤ 64 GB
+    - RAM if 64 < RAM ≤ 256 GB
+    - Not supported if RAM > 256 GB
+    """
+    free_tool = node.tools[Free]
+    df_tool = node.tools[Df]
+
+    # Get total memory in GB
+    total_memory_gb = float(free_tool.get_total_memory_gb())
+
+    # Skip hibernation check for VMs with > 256 GB RAM
+    if total_memory_gb > 256:
+        raise SkippedException(
+            f"Hibernation is not supported for VMs with RAM > 256 GB. "
+            f"Current RAM: {total_memory_gb:.2f} GB"
+        )
+
+    # Calculate required disk space based on RAM size
+    if total_memory_gb <= 8:
+        required_space_gb = 2 * total_memory_gb
+        formula = "2*RAM"
+    elif total_memory_gb <= 64:
+        required_space_gb = 1.5 * total_memory_gb
+        formula = "1.5*RAM"
+    else:
+        required_space_gb = total_memory_gb
+        formula = "RAM"
+
+    # Add 20% buffer for filesystem overhead, defragmentation, and safety margin
+    required_space_with_buffer = required_space_gb * 1.2
+
+    root_partition = df_tool.get_partition_by_mountpoint("/", force_run=True)
+    assert root_partition is not None, "Unable to determine root partition disk space"
+
+    # available_blocks is in 1K blocks, convert to GB
+    available_space_gb = root_partition.available_blocks / 1024 / 1024
+
+    if available_space_gb < required_space_with_buffer:
+        raise LisaException(
+            f"Insufficient disk space for hibernation. "
+            f"Memory size: {total_memory_gb:.2f} GB, "
+            f"Available space: {available_space_gb:.2f} GB, "
+            f"Required space: {required_space_gb:.2f} GB ({formula}), "
+            f"Recommended (20% buffer): {required_space_with_buffer:.2f} GB. "
+            f"Please increase 'osdisk_size_in_gb'."
```

Comment:
raise LisaException(
            "Insufficient disk space for hibernation. "
            f"Memory size: {total_memory_gb:.2f} GB, "
            f"Available space: {available_space_gb:.2f} GB, "
            f"Required space: {required_space_gb:.2f} GB ({formula}), "
            f"Recommended (20% buffer): {required_space_with_buffer:.2f} GB. "
            "Please increase 'osdisk_size_in_gb'."